Test/Test_IMU.c

```diff
@@ -0,0 +1,331 @@
+    3: It is a bit difficult to test since you would have to be moving your laptop. Since the magnetometer is also 
+    initialized, we will test to see if we get readings from that (So get a fridge magnet and wave it around the 
+    IMU and check to see if the data changes). According to the datasheet, it uses the magnetometer to read gravity
+    as well so there should be some data already being sent.
+    4: If that works, then move the IMU around very quickly. This should change the accelerometer and gyro readings.
+    5: According to the data collected, determine if we should initialize the IMU in another mode. (NDOF mode is 
+    one to consider. Page 25 of datasheet)
+    6: If readings are wrong, use calibration registers to make them more accurate. (Registers 0x55-0x6A)
+*/
+I2C_HandleTypeDef hi2c1;
+UART_HandleTypeDef huart3;
+
+CAN_HandleTypeDef hcan1;
+
+I2C_HandleTypeDef hi2c2;
+
+SPI_HandleTypeDef hspi5;
+
+UART_HandleTypeDef huart1;
+
+HAL_StatusTypeDef error;
+
+IMUCalibData_t Calib_data;
+
+
+//These defines are not in IMU.h because they are only necessary for debugging
+#define CALIB_STAT 0x35 //This register returns 0xFF if fully calibrated
+//If the below register returns a 1 anywhere, that means that it will only send values if the accelerometer
+//reads a change in data
+#define INT_STA 0x37 //We are not using low power mode, this should return all 0's (no interrupts)
+#define SYS_TRIGGER 0x3F //Setting this register to 0x01 starts a self test of the IMU
+//400 ms after the test starts, the below register will either remain 0(success) or return 3(failure)
+#define SYS_ERROR 0x3A 
+#define ST_RESULT 0x36 //This register will show a 0 for whichever device failed the test (pg. 50 of datasheet)
+#define OPR_MODE 0x3D
+
+
+
+void SystemClock_Config(void);
+static void MX_GPIO_Init(void);
+static void MX_I2C1_Init(void);
+static void MX_USART3_UART_Init(void);
+void StartDefaultTask(void *argument);
+void IMUTest(void * argument);
+
+osThreadId_t defaultTaskHandle;
+const osThreadAttr_t defaultTask_attributes = {
+  .name = "defaultTask",
+  .priority = (osPriority_t) osPriorityNormal,
+  .stack_size = 128 * 4
+};
+
+osThreadId_t IMUTestHandle;
+const osThreadAttr_t IMUTest_attributes = {
+  .name = "IMUTest",
+  .priority = (osPriority_t) osPriorityNormal,
+  .stack_size = 128 * 4
+};
+
+int main(void)
+{
+  HAL_Init();
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_I2C1_Init();
+  MX_USART3_UART_Init();
+
+  /* USER CODE END 2 */
+
+  /* Init scheduler */
+  osKernelInitialize();
+
+  defaultTaskHandle = osThreadNew(StartDefaultTask, NULL, &defaultTask_attributes);
+  IMUTestHandle = osThreadNew(IMUTest, NULL, &IMUTest_attributes);
+
+  /* Start scheduler */
+  osKernelStart();
+
+  while (1);
+}
+
+void IMUTest(void* argument){
+    //code taken from main.c in Core folder
+    osDelay(2000);
```

Comment:
Is there a reason for this long osDelay?

Reply:
Doesn't matter in this file